ddht/v5_1/alexandria/network.py

```diff
@@ -153,6 +161,242 @@ async def find_nodes(
     async def recursive_find_nodes(self, target: NodeID) -> Tuple[ENRAPI, ...]:
         return await common_recursive_find_nodes(self, target)
 
+    async def get_content_proof(
+        self,
+        node_id: NodeID,
+        *,
+        hash_tree_root: Hash32,
+        content_key: ContentKey,
+        start_chunk_index: int,
+        max_chunks: int,
+        endpoint: Optional[Endpoint] = None,
+        request_id: Optional[bytes] = None,
+    ) -> Proof:
+        if endpoint is None:
+            endpoint = await self.network.endpoint_for_node_id(node_id)
+
+        response = await self.client.get_content(
+            node_id,
+            endpoint,
+            content_key=content_key,
+            start_chunk_index=start_chunk_index,
+            max_chunks=max_chunks,
+            request_id=request_id,
+        )
+        if response.payload.is_proof:
+            proof = Proof.deserialize(
+                data=response.payload.payload, sedes=content_sedes,
+            )
+        else:
+            proof = compute_proof(
```

Comment:
fyi this logic fork is not yet triggered in tests or benchmarks

Reply:
As measured by.... did you run code coverage or something?

Reply:
Ah, nevermind, I see it.

Reply:
Lower-tech :) while manually stepping through stuff, a log/debug was never executed